core/spec/models/spree/moss_taxes_spec.rb

```diff
@@ -0,0 +1,128 @@
+require 'spec_helper'
+
+describe Spree::TaxRate, :type => :model do
+  context '#adjust with masala' do
+    let(:germany) { create :country, name: "Germany" }
+    let(:india) { create :country, name: "India" }
+    let(:france) { create :country, name: "France" }
+    let(:france_zone) { create :zone_with_country, name: "France Zone" }
+    let(:germany_zone) { create :zone_with_country, name: "Germany Zone", default_tax: true }
+    let(:india_zone) { create :zone_with_country, name: "India" }
+    let(:moss_category) { Spree::TaxCategory.create(name: "Digital Goods") }
+    let(:normal_category) { Spree::TaxCategory.create(name: "Analogue Goods") }
+    let(:eu_zone) { create(:zone, name: "EU" ) }
+
+    let!(:german_vat) do
+      Spree::TaxRate.create(
+        name: "German VAT",
+        amount: 0.19,
+        calculator: Spree::Calculator::DefaultTax.create,
+        tax_category: moss_category,
+        zone: germany_zone,
+        included_in_price: true
+      )
+    end
+    let!(:french_vat) do
+      Spree::TaxRate.create(
+        name: "French VAT",
+        amount: 0.25,
+        calculator: Spree::Calculator::DefaultTax.create,
+        tax_category: moss_category,
+        zone: france_zone,
+        included_in_price: true
+      )
+    end
+    let!(:eu_vat) do
+      Spree::TaxRate.create(
+        name: "EU_VAT",
+        amount: 0.19,
+        calculator: Spree::Calculator::DefaultTax.create,
+        tax_category: normal_category,
+        zone: eu_zone,
+        included_in_price: true
+      )
+    end
+
+    let(:download) { create(:product, tax_category: moss_category, price: 84.03) }
+    let(:tshirt) { create(:product, tax_category: normal_category, price: 84.03) }
+    let(:order) { Spree::Order.create }
+
+    before do
+      germany_zone.zone_members.create(zoneable: germany)
+      france_zone.zone_members.create(zoneable: france)
+      india_zone.zone_members.create(zoneable: india)
+      eu_zone.zone_members.create(zoneable: germany)
+      eu_zone.zone_members.create(zoneable: france)
+    end
+
+    context 'a download' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.